File: C_Programming/Competitive_Programming/CP_017.c
//////////////////////////////////////////////////////////////////
//
//  File Name    : Display_Factors_In_Reverse_Order
//
//  Description  : This program accepts a number from the user
//                 and displays all its factors in decreasing
//                 order, excluding the number itself.
//
//  Example:
//      Input  : 12
//      Output : 6 4 3 2 1
//
//      Input  : 13
//      Output : 1
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

void FactRev(int iNo)
{
    int i = 0;

    for(i = iNo / 2; i > 1; i--)
    {
        if(iNo % i == 0)
        {
            printf("%d\t", i);
        }        
    }
}
int main()
{
    int iValue = 0;

    printf("Enter Number : \n");
    scanf("%d", &iValue);

    FactRev(iValue);

    return 0;
}

File: Java_Programming/Competitive_Programming/CP_017.java
//////////////////////////////////////////////////////////////////
//
//  File Name    : Display_Digits_In_Reverse_Order
//
//  Description  : This program accepts a number and
//                 displays its digits in reverse
//                 order.
//
//  Example:
//      Input  : 1234
//      Output :
//          4
//          3
//          2
//          1
//
//////////////////////////////////////////////////////////////////

class Logic
{
    void printReverse(int iNum)
    {
        int iRev = 0;

        while(iNum > 0)
        {
            iRev = iNum % 10;
            System.out.println(iRev);
            iNum = iNum / 10;
        }
    }
}

class CP_017
{
    public static void main(String A[])
    {
        Logic lobj = new Logic();

        lobj.printReverse(10);
    }
}